src/Raven.Server/Documents/Replication/OutgoingReplicationHandler.cs

```diff
@@ -425,6 +332,225 @@ void HandleException(Exception e)
             }
         }
 
+        private void Replicate()
+        {
+            DateTime nextReplicateAt = default(DateTime);
+            var documentSender = new ReplicationDocumentSender(_stream, this, _log);
+
+            while (_cts.IsCancellationRequested == false)
+            {
+                while (_database.Time.GetUtcNow() > nextReplicateAt)
+                {
+                    if (_parent.DebugWaitAndRunReplicationOnce != null)
+                    {
+                        _parent.DebugWaitAndRunReplicationOnce.Wait(_cts.Token);
+                        _parent.DebugWaitAndRunReplicationOnce.Reset();
+                    }
+
+                    var sp = Stopwatch.StartNew();
+                    var stats = _lastStats = new OutgoingReplicationStatsAggregator(_parent.GetNextReplicationStatsId(), _lastStats);
+                    AddReplicationPerformance(stats);
+                    AddReplicationPulse(ReplicationPulseDirection.OutgoingBegin);
+
+                    try
+                    {
+                        using (var scope = stats.CreateScope())
+                        {
+                            try
+                            {
+                                if (Destination is InternalReplication dest)
+                                {
+                                    _parent.EnsureNotDeleted(dest.NodeTag);
+                                }
+
```

Comment:
Probably need to validate that this is still valid here. 
Need to revalidate if the pull replication def changed (especially if just a cert was removed)